zipkin-common/src/test/scala/com/twitter/zipkin/common/SpanTest.scala

```diff
@@ -31,6 +31,17 @@ class SpanTest extends FunSuite {
       BinaryAnnotation("sa", true, Some(Endpoint(1, 2, "sa"))),
       BinaryAnnotation("lc", "foo", Some(Endpoint(1, 2, "lc")))
     ))
+  val spanWithEmptyServiceNames = Span(12345, "methodcall", 666, None, None, None,
+    List(
+      Annotation(1, "cs", Some(Endpoint(1, 2, "cli"))),
+      Annotation(1, "sr", Some(Endpoint(1, 2, "srv"))),
+      Annotation(1, "ss", Some(Endpoint(1, 2, "srv"))),
+      Annotation(1, "cr", Some(Endpoint(1, 2, "cli")))
+    ),
+    List(
+      BinaryAnnotation("ca", true, Some(Endpoint(1, 2, ""))), // empty service name
```

Comment:
interesting use case.. basically this is when you know the IP/port, but not its name (pretty common, esp w/ X-Forwarded-For)